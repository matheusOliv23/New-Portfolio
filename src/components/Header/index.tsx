const menu = [
  { id: 1, name: "Home", route: "#home" },
  { id: 2, name: "Sobre Mim", route: "#about" },
  { id: 3, name: "Projetos", route: "#projects" },
  { id: 4, name: "Habilidades", route: "#skills" }
]

export default function Header() {
  return (
    <header className="flex w-full border-b-[0.3px] border-[#ffffff14] p-4 px-6 justify-end">
      <nav>
        <ul className="flex items-center gap-9 justify-between">
          {menu.map((item) => (
            <li className="hover:text-primary cursor-pointer" key={item.id}>
              {item.name}
            </li>
          ))}
        </ul>
      </nav>
    </header>
  )
}
