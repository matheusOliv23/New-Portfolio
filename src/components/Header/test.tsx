import { render, screen } from "@testing-library/react"

import Header from "./index"

const { getByText } = screen

describe("<Header />", () => {
  it("should render by default", () => {
    render(<Header />)

    const homeLink = getByText("Home")
    const aboutLink = getByText("Sobre Mim")
    const projectsLink = getByText("Projetos")
    const skillsLink = getByText("Habilidades")

    expect(homeLink).toBeInTheDocument()
    expect(aboutLink).toBeInTheDocument()
    expect(projectsLink).toBeInTheDocument()
    expect(skillsLink).toBeInTheDocument()
  })

  it("should match snapshot", () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
