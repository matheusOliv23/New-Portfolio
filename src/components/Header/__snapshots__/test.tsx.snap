// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <header
    class="flex w-full border-b-[0.3px] border-[#ffffff14] p-4 px-6 justify-end"
  >
    <nav>
      <ul
        class="flex items-center gap-9 justify-between"
      >
        <li
          class="hover:text-primary cursor-pointer"
        >
          Home
        </li>
        <li
          class="hover:text-primary cursor-pointer"
        >
          Sobre Mim
        </li>
        <li
          class="hover:text-primary cursor-pointer"
        >
          Projetos
        </li>
        <li
          class="hover:text-primary cursor-pointer"
        >
          Habilidades
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
